### PR TITLE
Fixes 3321

### PR DESCRIPTION
**Pull request by sorvell**

### CODE REVIEW

File: src/lib/dom-api-shady.html [hunk 1]
```diff
@@ -81,6 +82,11 @@
         } else {
           TreeApi.Composed.appendChild(container, node);
         }
+      // if a fragment provoked distribution, clean up it's actual dom.
+      } else if (isFragment) {
```
Comment (azakus) [inline src/lib/dom-api-shady.html]: Shouldn't this just be an `else`?

Comment (sorvell) [inline src/lib/dom-api-shady.html]: No, this is special handling specifically for fragments. When fragments are added to dom, all of their children are added.

Comment (azakus) [inline src/lib/dom-api-shady.html]: Ah my bad, the multiple `if`s next to each other was confusing, and I assumed the `else if (isFragment)` was part of the `if (!isFragment)` flow.

File: test/unit/polymer-dom-content.html [hunk 2]
```diff
@@ -1396,8 +1430,33 @@
...
+      frag.appendChild(document.createElement('div'));
+      Polymer.dom(x).appendChild(frag);
+      Polymer.dom.flush();
+      assert.equal(frag.childNodes.length, 0);
+    });
+
   });
 
+  
+
```
Comment (azakus) [inline test/unit/polymer-dom-content.html]: Leftover extra lines?

**Comment by azakus** [comment]
LGTM

**Comment by kevinpschaaf** [comment]
LGTM

**Pull request merged**